e2e/playwright/snapshot-tests.spec.ts

```diff
@@ -479,6 +479,69 @@ test('Draft segments should look right', async ({ page, context }) => {
   })
 })
 
+test('Rectangles should look right', async ({ page, context }) => {
+  const u = getUtils(page)
+  await page.setViewportSize({ width: 1200, height: 500 })
+  const PUR = 400 / 37.5 //pixeltoUnitRatio
+  await page.goto('/')
+  await u.waitForAuthSkipAppStart()
+  await u.openDebugPanel()
+
+  await expect(
+    page.getByRole('button', { name: 'Start Sketch' })
+  ).not.toBeDisabled()
+  await expect(page.getByRole('button', { name: 'Start Sketch' })).toBeVisible()
+
+  // click on "Start Sketch" button
+  await u.clearCommandLogs()
+  await u.doAndWaitForImageDiff(
+    () => page.getByRole('button', { name: 'Start Sketch' }).click(),
+    200
+  )
+
+  // select a plane
+  await page.mouse.click(700, 200)
+
+  await expect(page.locator('.cm-content')).toHaveText(
+    `const part001 = startSketchOn('-XZ')`
+  )
+
+  await page.waitForTimeout(300) // TODO detect animation ending, or disable animation
+
+  const startXPx = 600
+
+  // Equip the rectangle tool
+  await page.getByRole('button', { name: 'Line' }).click()
+  await page.getByRole('button', { name: 'Rectangle' }).click()
+
+  // Draw the rectangle
+  await page.mouse.click(startXPx + PUR * 20, 500 - PUR * 30)
+  await page.mouse.click(startXPx + PUR * 10, 500 - PUR * 10)
+  await page.waitForTimeout(100)
+
+  await u.closeDebugPanel()
+
+  // Ensure the rectangle is drawn by checking the KCL code
+  await expect(page.locator('.cm-content')).toHaveText(
+    `const part001 = startSketchOn('-XZ')
+      |> startProfileAt([18.2, 5.98], %)
+      |> angledLine([180, 9.14], %, 'rectangleSegmentA001')
+      |> angledLine([
+          segAng('rectangleSegmentA001', %) + 90,
+          18.2
+        ], %, 'rectangleSegmentB001')
+      |> angledLine([
+          segAng('rectangleSegmentA001', %),
+          -segLen('rectangleSegmentA001', %)
+        ], %, 'rectangleSegmentC001')
+      |> close(%)`
+  )
+
+  await expect(page).toHaveScreenshot({
+    maxDiffPixels: 100,
+  })
+})
```

Comment:
Trying to think about the philosophy of when we use snap shots, and I think it should be, "verify things visually when we can verify it in other ways".

And to that end asserting that the code here is something very specific makes me think that we should not need a screen shot, since we should be pretty confident that a simple sketch code like this will show up correctly in the app.

HOWEVER,

The draft lines before the second click might need verifying visually because there's no code in the app for that, and a check for "is the draft rectangle animating correctly?"

So maybe a better test is 
```typescript
await page.mouse.click(startXPx + PUR * 20, 500 - PUR * 30)
await page.mouse.move(startXPx + PUR * 10, 500 - PUR * 10, {steps: 5})
// take screenshot here!
await page.mouse.click(startXPx + PUR * 10, 500 - PUR * 10)

await expect(page.locator('.cm-content')).toHaveText( //...
```

Food for thought, up to you.

Reply:
Great point, I'll make this change